src/app/modules/color-scheme/color-bar/color-bar.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { ColorScheme } from '../color-schemes';
+
+/**
+ * Color bar component for the color scheme popup and color scheme selector
+ */
+@Component({
+  selector: 'ccf-color-bar',
+  templateUrl: './color-bar.component.html',
+  styleUrls: ['./color-bar.component.scss']
+})
+export class ColorBarComponent implements OnInit {
+
+  /**
+   * Color scheme to be displayed in the color bar
+   */
+  @Input() colorScheme: ColorScheme;
+
+  @Input() selected = false;
+
+  /**
+   * Emits the newly selected color
+   */
+  @Output() colorChange = new EventEmitter<string | undefined>();
+
+  /**
+   * Enables dynamic styling for gradient bars
+   */
+  gradientstyle: string;
+
+  selectedColorIndex = 0;
+
+  /**
+   * Sets the gradient style (when applicable) on load
+   */
+  ngOnInit() {
+    this.gradientstyle = this.colorScheme.type === 'gradient' ? `linear-gradient(to right, ${this.gradientColorString()})` : 'none';
+  }
+
+  gradientColorString() {
+    const result : string[] = [];
+    for (let i = 0; i < this.colorScheme.colors.length; i++) {
```

Comment:
When you find yourself repeating accesses to a variable assign it to a constant in the function. You can even use destructuring. I.e.
`const { colors, positions } = this.colorScheme;`